kona/src/main.rs

```diff
@@ -0,0 +1,326 @@
+                    let tip = info_from_header(&committed_chain.tip().block);
+                    self.chain_provider.commit(committed_chain);
+
+                    if tip.number >= self.cfg.genesis.l1.number {
+                        debug!(target: "kona", "Chain synced to rollup genesis with L1 block number: {}", tip.number);
+                        break Ok(tip);
+                    } else {
+                        trace!(target: "kona", "Chain not yet synced to rollup genesis. L1 block number: {}", tip.number);
+                    }
+
+                    if let Err(err) = self.ctx.events.send(ExExEvent::FinishedHeight(tip.number)) {
+                        bail!("Critical: Failed to send ExEx event: {:?}", err);
+                    }
+                }
+            }
+        }
+    }
+
+    /// Starts the Kona Execution Extension loop.
+    pub async fn start(mut self) -> Result<()> {
+        // Step 1: Wait for the L2 origin block to be available
+        let l2_genesis_l1_block = self.wait_for_l2_genesis_l1_block().await?;
+
+        // Step 2: Initialize the derivation pipeline with the L2 origin block
+        let mut pipeline = self.init_pipeline(l2_genesis_l1_block).await;
+
+        // Step 3: Start the main loop
+        loop {
+            self.step_l2(&mut pipeline).await;
+
+            if self.should_advance_l2_block_cursor {
+                if let Err(err) = self.advance_l2_cursor().await {
+                    error!(target: "kona", ?err, "Failed to advance L2 block cursor");
+                }
+            }
+
+            if let Ok(notification) = self.ctx.notifications.try_recv() {
+                if let Err(err) = self.handle_exex_notification(notification, &mut pipeline).await {
+                    error!(target: "kona", ?err, "Failed to handle ExEx notification");
+                }
+            }
+
+            if self.ctx.notifications.is_closed() {
+                warn!(target: "kona", "ExEx notification channel closed, exiting");
+                break Ok(());
+            }
+        }
+    }
+}
+
+fn main() -> Result<()> {
+    Cli::<KonaArgsExt>::parse().run(|builder, args| async move {
+        let Some(cfg) = ROLLUP_CONFIGS.get(&args.l2_chain_id).cloned().map(Arc::new) else {
+            bail!("Rollup configuration not found for L2 chain id: {}", args.l2_chain_id);
+        };
+
+        let node = EthereumNode::default();
+        let kona = move |ctx| async { Ok(KonaExEx::new(ctx, args, cfg).await.start()) };
+        let handle = builder.node(node).install_exex("Kona", kona).launch().await?;
+        handle.wait_for_node_exit().await
+    })
+}
+
+/// Helper to extract block info from a sealed block
+fn info_from_header(block: &reth::primitives::SealedBlock) -> BlockInfo {
+    BlockInfo {
+        hash: block.hash(),
+        number: block.number,
+        timestamp: block.timestamp,
+        parent_hash: block.parent_hash,
+    }
+}
```

Comment:
Would it make sense to upstream a `From<reth::primitives::SealedBlock> for BlockInfo` into kona?

Reply:
Kona doesn't import reth anywhere (yet), not sure if it's worth to add the dep for this, wdyt?